packages/@aws-cdk/aws-ecs/lib/drain-hook/instance-drain-hook.ts

```diff
@@ -97,11 +97,21 @@ export class InstanceDrainHook extends cdk.Construct {
       actions: [
         'ecs:ListContainerInstances',
         'ecs:SubmitContainerStateChange',
-        'ecs:SubmitTaskStateChange',
-        'ecs:UpdateContainerInstancesState',
-        'ecs:ListTasks'
+        'ecs:SubmitTaskStateChange'
       ],
       resources: [props.cluster.clusterArn]
     }));
+
+    // Restrict to the container-instance operations to the ECS Cluster
+    fn.addToRolePolicy(new iam.PolicyStatement({
+        actions: [
+          'ecs:UpdateContainerInstancesState',
+          'ecs:ListTasks'
+        ],
+        conditions: {
+            ArnEquals: {'ecs:cluster': props.cluster.clusterArn}
+        },
+        resources: ['*']
```

Comment:
Just making sure: does it make sense to use "*" here and not a specific resource?

Reply:
It does because otherwise, the resource would need to be the container instance id, but it doesn't make sense in the case of instances that are scaled up or down, as instance ids can change frequently. 

As per https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-supported-iam-actions-resources.html, we used the ecs:cluster condition key instead.